saplings/circuits/src/components/circuitsTable/TableHeader.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { Circuit } from '../../data/processCircuits';
+
+const sortCircuits = (circuits, action) => {
+  const order = action.orderAsc ? -1 : 1;
+  switch (action.sortBy) {
+    case 'comments': {
+      const sorted = circuits.sort((circuitA, circuitB) => {
+        if (circuitA.comments === 'N/A' && circuitB.comments !== 'N/A') {
+          return 1; // 'N/A should always be at the bottom'
+        }
+        if (circuitA.comments !== 'N/A' && circuitB.comments === 'N/A') {
+          return -1; // 'N/A should always be at the bottom'
+        }
+        if (circuitA.comments < circuitB.comments) {
+          return order;
+        }
+        if (circuitA.comments > circuitB.comments) {
+          return -order;
+        }
+        return 0;
+      });
+
+      return sorted;
+    }
+    case 'circuitID': {
+      const sorted = circuits.sort((circuitA, circuitB) => {
+        if (circuitA.id < circuitB.id) {
+          return order;
+        }
+        if (circuitA.id > circuitB.id) {
+          return -order;
+        }
+        return 0;
+      });
+
+      return sorted;
+    }
+    case 'serviceCount': {
+      const sorted = circuits.sort((circuitA, circuitB) => {
+        if (circuitA.roster.length < circuitB.roster.length) {
+          return order;
+        }
+        if (circuitA.roster.length > circuitB.roster.length) {
+          return -order;
+        }
+        return 0;
+      });
+
+      return sorted;
+    }
+    case 'managementType': {
+      const sorted = circuits.sort((circuitA, circuitB) => {
+        if (circuitA.managementType < circuitB.managementType) {
+          return order;
+        }
+        if (circuitA.managementType > circuitB.managementType) {
+          return -order;
+        }
+        return 0;
+      });
+
+      return sorted;
+    }
+    default:
+      return circuits;
+  }
+};
+
+const TableHeader = ({ dispatch, circuits }) => {
+  const [sorted, setSortedAsc] = useState({ asc: false, field: '' });
```

Comment:
I think the naming on these consts is a bit confusing. The typical convention is to have `[x, useX]`, so it is very clear what is being updated. I'm also not 100% sure what 'asc' is, so just spelling it out probably wouldn't hurt for maximum clarity.

Reply:
It would probably make sense to move this to the product reducer as well, since the reducer is really what is managing the sorted circuits.

Reply:
The convention I noticed by reading the [React docs](https://reactjs.org/docs/hooks-reference.html#usestate) was something like: 
```
const [state, setState] = useState(initialState);
```

That said, I agree that my naming was not very clear and I updated the it. Hopefully, it's better now.

Reply:
> It would probably make sense to move this to the product reducer as well, since the reducer is really what is managing the sorted circuits.

I don't think it makes sense to move this part to the circuits reducer. I think this is more about the component current state then really relevant to the whole circuits state.  No other component cares about how the circuits are sorted. This would only bloat the circuits reducer, I think.